### PR TITLE
Added commands for fetching incidents by the incident types and fetch…

### PR DESCRIPTION
**Pull request by ankita-sharma2019**
… incident details

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Armorblox/Integrations/Armorblox/Armorblox.py [hunk 1]
```diff
@@ -124,54 +135,42 @@ def get_remediation_action(client, incident_id):
     return CommandResults(outputs_prefix='Armorblox.Threat', outputs=contxt)
 
 
-def fetch_incidents_command(client):
+def fetch_incidents_command(client):  # pragma: no coverage
     last_run = demisto.getLastRun()
-    start_time: Any
-    # pageToken fetched from demisto lastRun
-    pageToken = int()
+    demisto.debug(str(last_run))
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/Armorblox.py]: Is this debug statement needed?

File: Packs/Armorblox/Integrations/Armorblox/Armorblox.py [hunk 2]
```diff
@@ -124,54 +135,42 @@ def get_remediation_action(client, incident_id):
...
-        if dt > int(start_time):
-            curr_incident = {'rawJSON': json.dumps(incident), 'details': json.dumps(incident)}
-            last_time = dt
-            incidents.append(curr_incident)
+        curr_incident = {'rawJSON': json.dumps(incident), 'details': json.dumps(incident)}
+        incidents.append(curr_incident)
     # Save the next_run as a dict with the start_time key to be stored
-    demisto.setLastRun({'start_time': str(last_time), 'pageToken': pageToken})
+    demisto.setLastRun({'last_fetch_time': last_fetch_time})
+    demisto.debug(str(len(incidents)))
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/Armorblox.py]: debug here too

File: Packs/Armorblox/Integrations/Armorblox/Armorblox_test.py [hunk 3]
```diff
@@ -96,23 +96,23 @@ def test_get_incidents_list(requests_mock):
...
-    requests_mock.get(url + '?orderBy=ASC&pageSize=1', json=incidents_response)
-    requests_mock.get(url + '?orderBy=ASC', json=incidents_response)
-    # get message ids function
-    mock_response_for_incident_id = util_load_json("test_data/test_response_for_6484.json")
-    requests_mock.get(url + '/6484', json=mock_response_for_incident_id)
-    client = Client(api_key=API_KEY, instance_name=TENANT_NAME)
-    response = fetch_incidents_command(client)
-    assert ('rawJSON' in response[0].keys()) is True
-    assert ('details' in response[0].keys()) is True
+# def test_fetch_incidents_command(requests_mock):
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/Armorblox_test.py:101]: Why is this all commented out? What do we want to do here?

**Review by urajat — APPROVED** [review]
Looks ok to me apart from a few minor comments

File: Packs/Armorblox/Integrations/Armorblox/Armorblox.py (hunk 1 above)
Comment (ankita-sharma2019) [reply Packs/Armorblox/Integrations/Armorblox/Armorblox.py]: No, Will remove

File: Packs/Armorblox/Integrations/Armorblox/Armorblox.py (hunk 2 above)
Comment (ankita-sharma2019) [reply Packs/Armorblox/Integrations/Armorblox/Armorblox.py]: Will remove

File: Packs/Armorblox/Integrations/Armorblox/Armorblox_test.py (hunk 3 above)
Comment (ankita-sharma2019) [reply Packs/Armorblox/Integrations/Armorblox/Armorblox_test.py:101]: This method was to test the `fetch_incidents` method. 

We have changed the logic to store last_fetch_time. This has changed the URL  to `https://outbound-integrations.armorblox.io/api/v1beta1/organizations/outbound-integrations/incidents?orderBy=ASC&from_date=2022-11-20T10:40:00Z&to_date=2022-11-21T10:40:00Z` from `https://outbound-integrations.armorblox.io/api/v1beta1/organizations/outbound-integrations/incidents?orderBy=ASC&timeFilter=lastDay`
So, here now the URL is not static and the values for the query parameter values are calculated dynamically based on first_fetch (lastDay/last3Days) parameter. So, we cannot create a static URL to mock. Hence, we have commented this method. 

We use the incident list and incident get methods here. And, we do have separate test methods for these URLs.

File: Packs/Armorblox/Integrations/Armorblox/Armorblox.yml [hunk 4]
```diff
@@ -73,7 +66,145 @@ script:
...
+    description: Gets information about an object associated with an incident.
+    name: armorblox-get-object-details
+  - arguments:
+    - description: The incident for which senders information is needed
+      name: incident_id
+      required: true
+    description: Gets information about an incident's sender data.
+    name: armorblox-get-incident-senders
+  - arguments:
+    - description: The incident id for which action needs to be  updated
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/Armorblox.yml]:
```suggestion
    - description: The incident id for which action needs to be updated
```

File: Packs/Armorblox/Integrations/Armorblox/README.md [hunk 5]
```diff
@@ -1,38 +1,49 @@
 Armorblox is an API-based platform that stops targeted email attacks,
   protects sensitive data, and automates incident response.
-This integration was integrated and tested with version 4.3.0 of Armorblox
+This integration was integrated and tested with version xx of Armorblox
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/README.md]:
```suggestion
This integration was integrated and tested with version 6.1.0 of Armorblox
```

File: Packs/Armorblox/Integrations/Armorblox/test_data/test_get_abuse_incidents.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
+{
+    "incidents": [
+        {
+            "priority": "HIGH",
+            "tagged": false,
+            "users": [
+                {
+                    "name": "Admin",
+                    "email": "admin@armorblox.dev",
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/test_data/test_get_abuse_incidents.json]: I thought we used dummy email ids last time?

File: Packs/Armorblox/Integrations/Armorblox/test_data/test_get_abuse_incidents.json [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            ],
+            "resolution_state": "OPEN_INCIDENT_RESOLUTION_STATE",
+            "object_type": "CONTENT_MAIL",
+            "id": "63424",
+            "research_status": "TRUE_POSITIVE",
+            "app_name": "MICROSOFT_OUTLOOK",
+            "external_users": [
+                {
+                    "name": "Bobby Lyons",
+                    "email": "bobby.lyons.ablox@gmail.com",
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/test_data/test_get_abuse_incidents.json]: This one too

File: Packs/Armorblox/Integrations/Armorblox/test_data/test_get_dlp_incidents.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        ],
+        "resolution_state": "OPEN_INCIDENT_RESOLUTION_STATE",
+        "object_type": "CONTENT_MAIL",
+        "id": "77648",
+        "research_status": "TRUE_POSITIVE",
+        "app_name": "MICROSOFT_OUTLOOK",
+        "external_users": [
+            {
+                "name": "Guestethanz",
+                "email": "ethanz@securerockets.com",
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/test_data/test_get_dlp_incidents.json]: This one too

File: Packs/Armorblox/Integrations/Armorblox/test_data/test_get_incident_by_id.json [hunk 9]
```diff
@@ -0,0 +1,648 @@
...
+    "policy_names": [
+        "Abuse Report"
+    ],
+    "remediation_actions": [
+        "ALERT"
+    ],
+    "users": [
+        {
+            "user_name": "Admin",
+            "user_address": "admin@armorblox.dev",
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/test_data/test_get_incident_by_id.json]: Replace all occurrences of this with a fake one - please check what we used last time.

File: Packs/Armorblox/Integrations/Armorblox/test_data/test_get_object_details.json [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    "subject": "This is 1094PAYROLL_FRAUD test 2022-06-09 12:22",
+    "mail_status": "DELETED",
+    "mail_header": {},
+    "from_user": "Todd Adams",
+    "sent_date_time": "seconds:1654757427",
+    "object_type": "CONTENT_MAIL",
+    "body_in_global": true,
+    "received_mailbox": {
+        "name": "james pardue",
+        "email": "james@armorblox.dev"
```
Comment (urajat) [inline Packs/Armorblox/Integrations/Armorblox/test_data/test_get_object_details.json]: This one

**Review by urajat — CHANGES_REQUESTED** [review]
Looks ok apart from a few minor suggestions

File: Packs/Armorblox/Integrations/Armorblox/test_data/test_get_abuse_incidents.json (hunk 6 above)
Comment (SamhithaTatipalli) [reply Packs/Armorblox/Integrations/Armorblox/test_data/test_get_abuse_incidents.json]: Yes @urajat , we would be replacing these mail ids with dummy ones.

**Review by urajat — APPROVED** [review]

**Pull request merged**